include/cudaq/Optimizer/CodeGen/QIRFunctionNames.h

```diff
@@ -27,6 +27,8 @@ static constexpr const char QIRCphase[] = "__quantum__qis__cphase";
 static constexpr const char QIRReadResultBody[] =
     "__quantum__qis__read_result__body";
 
+static constexpr const char QIRCustomOp[] = "__quantum__qis__custom_unitary";
```

Comment:
Isn't the current check at line#550 sufficient? Sorry, I don't follow where to put the additional check...

Reply:
That check on line 550 allows `__quantum_` functions to be called because they are part of the QIR standard, but this new functions is - to my knowledge - not really supported by any QIR-enabled backend, so I think it should be disallowed.
So that if check should probably be updated to something like `if (!funcName.startswith("__quantum_") || funcName.equals(QIRCustomOp))`.

Reply:
Understood. Thanks. Updated in b7389a5cc01c92b9941f231e00269bc10e3be796